RocketChat/Rocket.Chat
# [FIX] User avatar reseting and getting random image

## Proposed changes (including videos or screenshots)


- fixes user avatar not being saved after editing the user profile issue
- fixes user avatar not getting another user picture due to database deletion error



## Issue(s)


Fixes #24100

## Steps to test or reproduce

1. As a user log in to the app
2. Click on Avatar/My account
3. Change the username
4. Click on `Save changes` -> Should not change the avatar image
6. Logout and create a new user on the same server
7. Repeat steps 1 to 4 -> Should not change the avatar image as well

## Further comments
